src/blockdata/transaction.rs

```diff
@@ -353,7 +353,7 @@ impl Transaction {
             // will result in the data written to the writer being hashed, however the correct
             // handling of the SIGHASH_SINGLE bug is to return the 'one array' - either implement
             // this behaviour manually or use `signature_hash()`.
-            writer.write(b"[not a transaction] SIGHASH_SINGLE bug")?;
+            let _ = writer.write(b"[not a transaction] SIGHASH_SINGLE bug")?;
```

Comment:
In dfff85352a33cd5a12ca9134d4077e632ea8e35f:

Arguably the right thing to do here is to use `write_all` ... I'm torn. It's possible in principle that some user is depending on this exact text being written reliably, though it's a bit of a stretch.

Reply:
Agree it should be `write_all()`

However, why do we write BS data instead of returning error?!